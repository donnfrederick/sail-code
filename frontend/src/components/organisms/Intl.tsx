import React, {
  PropsWithChildren,
  Dispatch,
  SetStateAction,
  useContext,
  useState
} from 'react'
import { IntlProvider } from 'react-intl'
import { messages, browserLocale, Locale } from 'utils/intl'

export type LocaleState = {
  locale: Locale
}

const initialState: LocaleState = {
  locale: browserLocale
}

const IntlStateContext = React.createContext<LocaleState>(initialState)

const SetLocaleStateContext = React.createContext<
  Dispatch<SetStateAction<LocaleState>>
>(() => {
  // empty block
})

export function useLocaleState() {
  return useContext(IntlStateContext)
}
export function useSetLocaleState() {
  return useContext(SetLocaleStateContext)
}

export function LocaleStateProvider(props: {
  initialState?: LocaleState
  children: React.ReactNode
}) {
  const [state, setState] = useState<LocaleState>(
    props.initialState || initialState
  )
  return (
    <IntlStateContext.Provider value={state}>
      <SetLocaleStateContext.Provider value={setState}>
        {props.children}
      </SetLocaleStateContext.Provider>
    </IntlStateContext.Provider>
  )
}

export const Intl = (props: PropsWithChildren<{}>) => {
  const localeState = useLocaleState()

  return (
    <LocaleStateProvider>
      <IntlProvider
        messages={messages[localeState.locale]}
        locale={localeState.locale}
        defaultLocale="en"
      >
        {props.children}
      </IntlProvider>
    </LocaleStateProvider>
  )
}
