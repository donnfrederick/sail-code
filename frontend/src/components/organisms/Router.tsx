import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
