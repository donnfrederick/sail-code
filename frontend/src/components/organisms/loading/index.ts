import Loading from 'components/organisms/loading/Loading'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  isFetchingConversation: boolean
  isFetchingLocations: boolean
  isFetchingNotification: boolean
  isFetchingOrganizations: boolean
  isFetchingStudentsMe: boolean
  isFetchingTeachersMe: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    isFetchingConversation: state.rootReducer.conversation.fetching > 0,
    isFetchingLocations: state.rootReducer.locations.isFetching,
    isFetchingNotification: state.rootReducer.notification.isFetching,
    isFetchingOrganizations: state.rootReducer.organizations.isFetching,
    isFetchingStudentsMe: state.rootReducer.students.isFetching,
    isFetchingTeachersMe: state.rootReducer.teachers.isFetching
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {}
)

export default enhancer(Loading)
