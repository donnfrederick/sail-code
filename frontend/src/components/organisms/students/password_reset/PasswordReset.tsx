import Button from 'components/atoms/students/Button'
import EmailInput from 'components/molecules/students/EmailInput'
import FailFetchingModal from 'components/organisms/students/modal_contents/fail_fetching'
import * as StudentsModels from 'models/students'
import * as React from 'react'
import styled from 'styled-components'
import resolvePath from 'utils/resolvePath'

interface Props {
  error: any
  info: StudentsModels.Info
  sentPasswordResetRequest: boolean
  openModal(): void
  postPassword(email: string): void
  register(info: StudentsModels.Info): void
  setModalContents(contents: JSX.Element): void
}

export default (props: Props) => {
  const {
    error,
    info,
    sentPasswordResetRequest,
    openModal,
    postPassword,
    register,
    setModalContents
  } = props

  return sentPasswordResetRequest ? (
    <Container>
      <Heading>
        {
          "We've sent an email to you.\nClick the link in the email to reset your password."
        }
      </Heading>
      <ButtonContainer>
        <Button
          type="blue"
          text="Back"
          width={494}
          height={88}
          link={resolvePath.page('students', 'signin')}
        />
      </ButtonContainer>
    </Container>
  ) : (
    <Container>
      <Heading>{'Please enter your email'}</Heading>
      <EmailInput error={error} info={info} register={register} />
      <ButtonContainer>
        <Button
          type="blue"
          text="Send"
          width={494}
          height={88}
          isActive={info.email.length > 0}
          onClick={async () => {
            try {
              await postPassword(info.email)
            } catch (error) {
              // tslint:disable-next-line
              console.error(error)
              setModalContents(<FailFetchingModal />)
              openModal()
            }
          }}
        />
      </ButtonContainer>
    </Container>
  )
}

const Container = styled.div`
  width: 100%;
`

const Heading = styled.div`
  margin-bottom: 208px;
  font-size: 40px;
  font-weight: 500;
  line-height: 1.3;
  text-align: center;
  white-space: pre-wrap;
  color: #405766;
`

const ButtonContainer = styled.div`
  display: flex;
  justify-content: center;
  position: absolute;
  bottom: 56px;
  left: 0;
  width: 100%;
`
