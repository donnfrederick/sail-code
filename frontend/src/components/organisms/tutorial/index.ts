import Tutorial from 'components/organisms/tutorial/Tutorial'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as TutorialActions from 'reducers/tutorial'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  shouldShow: boolean
  showConfirmation: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    shouldShow: state.rootReducer.tutorial.shouldShow,
    showConfirmation: state.rootReducer.teachers.showConfirmation
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    hide: TutorialActions.hide,
    show: TutorialActions.show
  }
)

export default enhancer(Tutorial)
