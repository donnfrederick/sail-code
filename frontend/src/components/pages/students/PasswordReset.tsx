import StudentsPasswordReset from 'components/templates/students/PasswordReset'
import * as React from 'react'

export default () => {
  return <StudentsPasswordReset />
}
