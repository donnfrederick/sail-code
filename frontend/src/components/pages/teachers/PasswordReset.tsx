import TeachersPasswordReset from 'components/templates/teachers/PasswordReset'
import * as React from 'react'

export default () => {
  return <TeachersPasswordReset />
}
