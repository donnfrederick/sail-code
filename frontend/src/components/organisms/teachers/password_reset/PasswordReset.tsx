import Button from 'components/atoms/teachers/Button'
import InputMailAddress from 'components/atoms/teachers/InputMailAddress'
import FailFetchingModal from 'components/organisms/teachers/modal_contents/fail_fetching'
import * as TeachersModels from 'models/teachers'
import * as React from 'react'
import styled from 'styled-components'
import resolvePath from 'utils/resolvePath'

interface Props {
  error: any
  info: TeachersModels.Info
  sentPasswordResetRequest: boolean
  openModal(): void
  postPassword(email: string): void
  register(info: TeachersModels.Info): void
  setModalContents(contents: JSX.Element): void
}

interface ButtonProps {
  hasSingleButton: boolean
}

export default (props: Props) => {
  const {
    error,
    info,
    sentPasswordResetRequest,
    openModal,
    postPassword,
    register,
    setModalContents
  } = props

  return sentPasswordResetRequest ? (
    <Container>
      <Heading>{'送信が完了しました'}</Heading>
      <Text>
        {
          'パスワード再設定用のメールを送信しました。\n登録しているメールアドレスをご確認ください。'
        }
      </Text>
      <ButtonContainer hasSingleButton={true}>
        <Button
          type="white"
          text="戻る"
          width={384}
          height={112}
          link={resolvePath.page('teachers', 'signin')}
        />
      </ButtonContainer>
    </Container>
  ) : (
    <Container>
      <Heading>{'登録したメールアドレスを\n入力してください'}</Heading>
      <InputMailAddress error={error} info={info} register={register} />
      <ButtonContainer hasSingleButton={false}>
        <Button
          type="white"
          text="戻る"
          width={304}
          height={112}
          link={resolvePath.page('teachers', 'signin')}
        />
        <Button
          type="blue"
          text="送信"
          width={304}
          height={112}
          isActive={info.email.length > 0}
          onClick={async () => {
            try {
              await postPassword(info.email)
            } catch (error) {
              // tslint:disable-next-line
              console.error(error)
              setModalContents(<FailFetchingModal />)
              openModal()
            }
          }}
        />
      </ButtonContainer>
    </Container>
  )
}

const Container = styled.div`
  width: 100%;
`

const Heading = styled.div`
  margin-bottom: 152px;
  font-size: 40px;
  font-weight: 500;
  line-height: 1.3;
  text-align: center;
  white-space: pre-wrap;
  color: #405766;
`

const Text = styled.div`
  font-size: 28px;
  line-height: 1.71;
  text-align: center;
  white-space: pre-wrap;
  color: #405766;
`

const ButtonContainer = styled<ButtonProps, any>('div')`
  display: flex;
  justify-content: ${props =>
    props.hasSingleButton ? 'center' : 'space-between'};
  position: absolute;
  bottom: 56px;
  left: 80px;
  width: calc(100% - 80px * 2);
`
