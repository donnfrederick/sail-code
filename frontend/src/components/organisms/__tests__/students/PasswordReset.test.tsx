import Router from 'components/organisms/Router'
import PasswordReset from 'components/organisms/students/password_reset'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Intl } from 'components/organisms/Intl'

test('render <PasswordReset />', () => {
  const tree = renderer
    .create(
      <Intl>
        <Router>
          <PasswordReset />
        </Router>
      </Intl>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
