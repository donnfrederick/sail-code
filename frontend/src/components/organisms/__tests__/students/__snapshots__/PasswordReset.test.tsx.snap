// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <PasswordReset /> 1`] = `
<div
  className="sc-jTzLTM hDLiwJ"
>
  <div
    className="sc-fjdhpX gqbTsM"
  >
    Please enter your email
  </div>
  <div
    className="sc-htoDjs bXSIZO"
  >
    <div
      className="sc-gzVnrw cegWGq"
    >
      Email
    </div>
    <input
      className="sc-bZQynM kKwPOo"
      data-error={false}
      onInput={[Function]}
      placeholder="Email"
      type="email"
    />
    <div
      className="sc-EHOje gtuuwv"
    />
  </div>
  <div
    className="sc-jzJRlG chznXU"
  >
    <button
      className="sc-bdVaJa eEjsbL"
      data-active={false}
      fontSize={32}
      height={88}
      onClick={[Function]}
      type="blue"
      width={494}
    >
      Send
    </button>
  </div>
</div>
`;
